### PR TITLE
Add Template for EAP S2I build

### PR DESCRIPTION
**Pull request by jmesnil**
Signed-off-by: Jeff Mesnil <jmesnil@redhat.com>

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+...
+Successfully pushed image-registry.openshift-image-registry.svc:5000/demo/my-app-build-artifacts@sha256:01dc775bb76e00cbd8ccd31d4887e274ebcd6d2279aad39bc98a4fa749da974d
+Push successful
+----
+
+Then the actual application image will be built by using the EAP S2I Runtime image so that it contains only the runtime required to run the application:
+
+[source,bash]
+----
+$ oc logs -f bc/my-app-build-artifacts
```
Comment (jbliznak) [inline README.adoc]: Should this be the same command as in previous step and not the `oc logs -f bc/my-app` instead?

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+my-app   default-route-openshift-image-registry.apps-crc.testing/demo/my-app   latest   9 minutes ago
+----
+
+== Deploy an Application Image Using the EAP Operator
+
+Once the application image has been built, it can be deployed using the EAP Operator by creating a `WildFlyServer` resource:
+
+[source,yaml]
+----
+cat <<EOF | kubectl apply -f -
```
Comment (jbliznak) [inline README.adoc]: `kubectl` -> `oc` to be in sync with other commands here

File: eap-s2i-build.yaml [hunk 3]
```diff
@@ -0,0 +1,172 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: eap-s2i-build
+  xpaas: "1.0"
+  app.kubernetes.io/name: ${APPLICATION_IMAGE}
+message: A new ImageStream for your JBoss EAP based application has been created in your project.
+metadata:
+  annotations:
+    description: This template build an application image based on JBoss EAP from a Git repository using S2I.
```
Comment (jbliznak) [inline eap-s2i-build.yaml]: typo `builds`

File: eap-s2i-build.yaml [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    iconClass: icon-eap
+    openshift.io/display-name: Template to build JBoss EAP application using S2I
+    openshift.io/provider-display-name: Red Hat
+    tags: eap,javaee,java,jboss
+    template.openshift.io/documentation-url: https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/
+    template.openshift.io/long-description: This template build an image based on JBoss EAP using S2I.
```
Comment (jbliznak) [inline eap-s2i-build.yaml]: typo `builds`

File: README.adoc (hunk 1 above)
Comment (jmesnil) [reply README.adoc]: fixed, thanks

**Comment by jmesnil** [comment]
@jbliznak I've fixed all your comments, thanks

**Pull request merged**